### ISSUE TITLE
Why evaluate twice?

### ISSUE THREAD

**Issue by Shiweiliuiiiiiii** [sarthmit/Mod_Arch#1]
Hi,

Thanks a lot for releasing this great source code. I am wondering why you evaluate the model twice here. Did I miss something here?
 
eval_loss, eval_acc = eval_step()
val_loss, val_acc = eval_step()
https://github.com/sarthmit/Mod_Arch/blob/fd47bcae82f09dada0aca958f0c770c825651f4e/MHA/Classification/main.py#L184

Best,
Shiwei

**Comment by sarthmit** [comment]
Hey! Thanks for the interest!
We do one evaluation for validation loss/accuracy and one for test loss/accuracy; and save the final model as well as the one corresponding to the best validation performance. Of course in this setting since the data is generated on the fly, it doesn't make any difference but it would be useful if one were to extend this exact setting to finite-data regime.

Hope that clarifies it. Feel free to re-open the issue if it doesn't :)
